### PR TITLE
core/api: Add grpc-gateway api

### PR DESCRIPTION
**Pull request by demophoon**
This PR enables gRPC-Gateway on the http port of Waypoint server to add in an HTTP API for interfacing with Waypoint.

Most of this PR is autogenerated from the existing `server.swagger.json` file that is directly generated from our protos.

### CODE REVIEW

File: internal/server/http.go [hunk 1]
```diff
@@ -48,8 +50,17 @@ func newHttpServer(grpcServer *grpc.Server, ln net.Listener, opts *options) *htt
 	// gRPC server. This is used by the exec handler.
 	grpcAddr := opts.GRPCListener.Addr().String()
 
+	// Create grpc-gateway muxer
+	grpcMux := runtime.NewServeMux()
+	grpcOpts := []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())}
+	err := gen.RegisterWaypointHandlerFromEndpoint(opts.Context, grpcMux, "localhost:9701", grpcOpts)
+	if err != nil {
+		log.Error("Unable to register waypoint grpc gateway service")
```
Comment (briancain) [inline internal/server/http.go:58]: Hrm, I think we'd want to return this error too instead of only logging it?

**Review by evanphx — APPROVED** [review]
Innocently small once you remove the generated files!

File: internal/server/http.go (hunk 1 above)
Comment (demophoon) [reply internal/server/http.go]: Resolved in `c9a3fd348844a628157b0e0167822bb14764024d`

File: .changelog/4379.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+api: Enable HTTP API to talk to Waypoint
```
Comment (briancain) [inline .changelog/4379.txt:2]: We should make this more descriptive, something like

```suggestion
server: Enable gRPC-Gateway on the http port of Waypoint server to add in an HTTP API for interfacing
```

**Review by briancain — APPROVED** [review]

**Pull request merged**